### PR TITLE
[LifetimeChecker] Update the debug scope when changing insertion point.

### PR DESCRIPTION
**Pull request by dcci**
Another day, another debug info bug fixed.

<rdar://problem/36679700>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test and merge

File: lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp [hunk 1]
```diff
@@ -2436,6 +2436,7 @@ handleConditionalDestroys(SILValue ControlVariableAddr) {
     auto &Availability = CDElt.Availability;
 
     B.setInsertionPoint(Release);
+    B.setCurrentDebugScope(Release->getDebugScope());
```
Comment (adrian-prantl) [inline lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp]: This makes me wonder if we should make it illegal to set the insertion point without also doing something to the scope. Should we replace setInsertionPoint with setInsertionPointAndDebugScope() ?

Comment (adrian-prantl) [reply lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp]: @vedantk For the "let's improve SILBuilder ergonomics" task ^^

**Review by adrian-prantl — APPROVED** [review]

File: lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp (hunk 1 above)
Comment (adrian-prantl) [reply lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp]: @dcci: or should these all be fresh SILBuilderWithScopes?

Comment (dcci) [reply lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp]: Probably we should do something like that, let me think and let's chat further.

**Comment by dcci** [comment]
@vedantk Considering the amount of time I'm spending fixing all these violations, I'll buy you a drink if you fix the SILBuilder API ergonomics.

**Comment by dcci** [comment]
ugh, this is failing on Linux because there's no Foundation, let me adjust and run again.

**Comment by dcci** [comment]
@swift-ci test and merge

File: lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp (hunk 1 above)
Comment (vedantk) [reply lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp]: I proposed setInsertionPointAndDebugScope() for llvm's IRBuilder. The main sticking point was on how to update all existing uses of setInsertionPoint(). It's worth having a separate discussion about this for SILGen.

**Comment by dcci** [comment]
@swift-ci please test

**Review by vedantk — APPROVED** [review]
Thanks!

**Comment by dcci** [comment]
@swift-ci please test and merge

**Comment by dcci** [comment]
@swift-ci please test and merge

**Comment by dcci** [comment]
@swift-ci please test and merge

**Pull request merged**